integration/simulation/simulation.go

```diff
@@ -52,11 +53,11 @@ func RunSimulation(
 			genesis = true
 		}
 		// create a layer 2 node
-		agg := obscuro_node.NewAgg(common.NodeID(i), l2Cfg, nil, &l2Network, &stats, genesis)
+		agg := obscuro_node.NewAgg(common.NodeID(common2.BigToAddress(big.NewInt(int64(i)))), l2Cfg, nil, &l2Network, &stats, genesis)
```

Comment:
I think it would be cleaner if `NewAgg()` and `NewMiner()` generated a random address, rather than having one passed in.